### PR TITLE
13287 export certificates progress alert

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Show a waiting alert to the user, when tap on Next in Export-All-Certificates-Info, until PDF preparations are done. 
The user can cancel the process.

## Link to Jira
[https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13287](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13287)

## Screenshots
Screen Shot 2022-06-13 at 17 01 30

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewModel.swift [hunk 1]
```diff
@@ -54,47 +56,59 @@ class HealthCertificateExportCertificatesInfoViewModel {
...
-		vaccinationValueSetsProvider.latestVaccinationCertificateValueSets()
-			.sink(
-				receiveCompletion: { result in
-					switch result {
-					case .finished:
-						break
-					case .failure(let error):
-						if case CachingHTTPClient.CacheError.dataVerificationError = error {
-							Log.error("Signature verification error.", log: .vaccination, error: error)
+		// Delay to give user the chance to recognize the regarding info alert
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewModel.swift:59]:
```suggestion
		// Delay to give user a chance to see the content on info alert or cancel the process 
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift [hunk 2]
```diff
@@ -91,4 +98,46 @@ class HealthCertificateExportCertificatesInfoViewController: DynamicTableViewCon
...
+			handler: { [weak self] _ in
+				self?.viewModel.removeAllSubscriptions()
+				self?.onDismiss(true)
+			}
+		))
+
+		return alert
+	}()
+	
+	private func showGeneratePDFDataAlert() {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift:121]:
```suggestion
	private func showPDFDataAlert() {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift [hunk 3]
```diff
@@ -91,4 +98,46 @@ class HealthCertificateExportCertificatesInfoViewController: DynamicTableViewCon
...
+		generatePDFDataAlertController.message = AppStrings.HealthCertificate.ExportCertificatesInfo.generatePDFProgressMessageInitial
+		present(generatePDFDataAlertController, animated: true)
+	}
+	
+	private func hideGeneratePDFDataAlert(completion: @escaping CompletionVoid) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift:126]:
```suggestion
	private func hidePDFDataAlert(completion: @escaping CompletionVoid) {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift [hunk 4]
```diff
@@ -91,4 +98,46 @@ class HealthCertificateExportCertificatesInfoViewController: DynamicTableViewCon
...
+		generatePDFDataAlertController.dismiss(animated: true) {
+			completion()
+		}
+	}
+	
+	private func updateGeneratePDFDataAlertMessage(pageInProgress: Int, numberOfPages: Int) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Consent/HealthCertificateExportCertificatesInfoViewController.swift:132]:
```suggestion
	private func updatePDFDataAlertMessage(pageInProgress: Int, numberOfPages: Int) {
```

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**